packages/react-router/src/client/ReactRouterClerkProvider.tsx

```diff
@@ -0,0 +1,151 @@
+ */
+const awaitableNavigateRef: { current: ReturnType<typeof useAwaitableNavigate> | undefined } = { current: undefined };
+
+/**
+ * Internal type that includes the initial state prop that is passed to the ClerkProvider during SSR.
+ * This is a value that we pass automatically so it does not need to pollute the public API.
+ */
+type ClerkProviderPropsWithState = ReactRouterClerkProviderProps & {
+  clerkState?: ClerkState;
+};
+
+function ClerkProviderBase({ children, ...rest }: ClerkProviderPropsWithState): JSX.Element {
+  const awaitableNavigate = useAwaitableNavigate();
+  const isSpaMode = inSpaMode();
+
+  React.useEffect(() => {
+    awaitableNavigateRef.current = awaitableNavigate;
+  }, [awaitableNavigate]);
+
+  const { clerkState, ...restProps } = rest;
+  ReactClerkProvider.displayName = 'ReactClerkProvider';
+
+  if (!isSpaMode) {
+    assertValidClerkState(clerkState);
+  }
+
+  const {
+    __clerk_ssr_state,
+    __publishableKey,
+    __proxyUrl,
+    __domain,
+    __isSatellite,
+    __clerk_debug,
+    __signInUrl,
+    __signUpUrl,
+    __afterSignInUrl,
+    __afterSignUpUrl,
+    __signInForceRedirectUrl,
+    __signUpForceRedirectUrl,
+    __signInFallbackRedirectUrl,
+    __signUpFallbackRedirectUrl,
+    __clerkJSUrl,
+    __clerkJSVersion,
+    __telemetryDisabled,
+    __telemetryDebug,
+  } = clerkState?.__internal_clerk_state || {};
+
+  React.useEffect(() => {
+    if (!isSpaMode) {
+      warnForSsr(clerkState);
+    }
+  }, []);
+
+  React.useEffect(() => {
+    (window as any).__clerk_debug = __clerk_debug;
+  }, []);
+
+  const mergedProps = {
+    publishableKey: __publishableKey as any,
+    proxyUrl: __proxyUrl as any,
+    domain: __domain as any,
+    isSatellite: __isSatellite,
+    signInUrl: __signInUrl,
+    signUpUrl: __signUpUrl,
+    afterSignInUrl: __afterSignInUrl,
+    afterSignUpUrl: __afterSignUpUrl,
+    signInForceRedirectUrl: __signInForceRedirectUrl,
+    signUpForceRedirectUrl: __signUpForceRedirectUrl,
+    signInFallbackRedirectUrl: __signInFallbackRedirectUrl,
+    signUpFallbackRedirectUrl: __signUpFallbackRedirectUrl,
+    clerkJSUrl: __clerkJSUrl,
+    clerkJSVersion: __clerkJSVersion,
+    telemetry: {
+      disabled: __telemetryDisabled,
+      debug: __telemetryDebug,
+    },
+  };
+
+  return (
+    <ClerkReactRouterOptionsProvider options={mergedProps}>
+      <ReactClerkProvider
+        routerPush={(to: string) => awaitableNavigateRef.current?.(to)}
+        routerReplace={(to: string) => awaitableNavigateRef.current?.(to, { replace: true })}
+        initialState={__clerk_ssr_state}
+        sdkMetadata={SDK_METADATA}
+        {...mergedProps}
+        {...restProps}
+      >
+        {children}
+      </ReactClerkProvider>
+    </ClerkReactRouterOptionsProvider>
+  );
+}
+
+type ClerkReactRouterOptions = Partial<
+  Omit<ReactRouterClerkProviderProps, 'routerPush' | 'routerReplace' | 'clerkState'>
+>;
+
+// TODO: Remove "any" on loaderData type
```

Comment:
With some TS magic I think we can achieve that the `Route.LoaderArgs` is passed through to our component, right?